data/data_loader.py

```diff
@@ -32,24 +27,23 @@ class DataLoaderMultiAspect():
     """
     Data loader for multi-aspect-ratio training and bucketing
 
-    data_root: root folder of training data
+    image_train_items: list of `lImageTrainItem` objects
+    seed: random seed
     batch_size: number of images per batch
-    flip_p: probability of flipping image horizontally (i.e. 0-0.5)
     """
-    def __init__(self, data_root, seed=555, debug_level=0, batch_size=1, flip_p=0.0, resolution=512, log_folder=None):
-        self.data_root = data_root
-        self.debug_level = debug_level
-        self.flip_p = flip_p
-        self.log_folder = log_folder
+    def __init__(self, image_train_items: list[ImageTrainItem], seed=555, batch_size=1):
         self.seed = seed
         self.batch_size = batch_size
-        self.has_scanned = False
-        self.aspects = aspects.get_aspect_buckets(resolution=resolution, square_only=False)
-        
-        logging.info(f"* DLMA resolution {resolution}, buckets: {self.aspects}")
-        self.__prepare_train_data()
-        (self.rating_overall_sum, self.ratings_summed) = self.__sort_and_precalc_image_ratings()
-
+        # Prepare data
+        self.prepared_train_data = image_train_items
+        random.Random(self.seed).shuffle(self.prepared_train_data)
+        self.prepared_train_data = sorted(self.prepared_train_data, key=lambda img: img.caption.rating())
+        # Initialize ratings
+        self.rating_overall_sum: float = 0.0
+        self.ratings_summed: list[float] = []
+        for image in self.prepared_train_data:
+            self.rating_overall_sum += image.caption.rating()
+            self.ratings_summed.append(self.rating_overall_sum)
```

Comment:
Since there wasn't much left of `__prepare_train_data` and `__sort_and_precalc_image_ratings` is so small, and both are only used during construction, I moved them their contents here.